### PR TITLE
Updated Datepicker widget

### PR DESCRIPTION
**Pull request by mkotsollaris**

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -113,6 +181,25 @@ const validateAdditionalFields = (additionalFields) => {
   }
 };
 
+const parseSubmitTruliooDateFields = (obj) => {
+  Object.keys(obj).forEach((key) => {
+    if (key === DOB) {
+      const splitDate = obj[key].split('-');
+      const day = splitDate[2];
+      const month = splitDate[1];
+      const year = splitDate[0];
```
Comment (kateika) [inline src/actions/index.js]: const [year,month,day] = splitDate

File: src/actions/index.js [hunk 2]
```diff
@@ -113,6 +181,25 @@ const validateAdditionalFields = (additionalFields) => {
...
+      obj[DAY_OF_BIRTH] = day;
+      obj[MONTH_OF_BIRTH] = month;
+      obj[YEAR_OF_BIRTH] = year;
+      delete obj[DOB];
+    }
+    if (typeof obj[key] === 'object') {
```
Comment (kateika) [inline src/actions/index.js]: null is object as well

File: src/tests/unit/actions.test.js [hunk 3]
```diff
@@ -166,4 +166,8 @@ describe('Testing actions without detailed consents', () => {
       expect(expectedActions.length).toEqual(receivedActions.length);
     });
   });
+
+  it('getFields displays DOB as an unique field when given Day, Month and Year field combincation', () => {
+    // TODO
+  });
```
Comment (kateika) [inline src/tests/unit/actions.test.js]: not sure if you want to merge that to master

File: src/tests/unit/mock_payloads/formSubmitDocV.json [hunk 4]
```diff
@@ -1,7 +1,8 @@
 {
     "TruliooFields": {
         "PersonInfo": {
-            "FirstGivenName": "examoke"
+            "FirstGivenName": "examle",
```
Comment (kateika) [inline src/tests/unit/mock_payloads/formSubmitDocV.json]: is it example or what examle means?

Comment (mkotsollaris) [reply src/tests/unit/mock_payloads/formSubmitDocV.json]: good catch

File: src/actions/index.js [hunk 5]
```diff
@@ -36,11 +40,75 @@ const parseFields = (obj) => {
...
+    return;
+  }
+  const { required } = obj;
+  if (!required) {
+    return;
+  }
+  const containsDOB = containsDOBRequired(obj.required);
+  if (containsDOB) {
+    obj.required = required.filter((requiredField) => (requiredField !== DAY_OF_BIRTH
+      && requiredField !== MONTH_OF_BIRTH && requiredField !== YEAR_OF_BIRTH));
```
[the remark this answers is not in the document]
Comment (mkotsollaris) [reply src/actions/index.js]: What I basically do, is essentially, to replace `day` , `month` and `year` of birth with `DOB`

**Pull request merged**